### PR TITLE
OSDOCS-12307: CA Parameters Image registry config to customers

### PR DESCRIPTION
**Pull request by fmcdonal**
Version(s):
4.17

Issue:
https://issues.redhat.com/browse/OSDOCS-12307

Link to docs preview:
https://83499--ocpdocs-pr.netlify.app/openshift-rosa/latest/openshift_images/image-configuration-hcp.html

QE review:
- [ ] QE has approved this change.

### CODE REVIEW

File: modules/images-configuration-image-registry-settings-hcp.adoc [hunk 1]
```diff
@@ -96,17 +96,21 @@ Audit Log Forwarding:       Disabled
...
- - Allowed Registries for Import: <3>
-    - Domain Name:          <domain_name> <4>
-    - Insecure:             true <5>
+ - Allowed Registries:     <allowed_registry> <1> <2>
+ - Insecure Registries:    <insecure_registry> <3>
+ - Allowed Registries for Import: <4>
+    - Domain Name:          <domain_name> <5>
+    - Insecure:             true <6>
+ - Additional Trusted CA: <7>
+    - <ca_name> : REDACTED
```
Comment (chenz4027) [inline modules/images-configuration-image-registry-settings-hcp.adoc:105]: should be <registry_name>

File: modules/images-editing-image-registry-settings-hcp.adoc [hunk 2]
```diff
@@ -104,14 +104,18 @@ Audit Log Forwarding:       Disabled
...
- - Allowed Registries for Import: <3>
-    - Domain Name:          <domain_name> <4>
-    - Insecure:             true <5>
+ - Allowed Registries: <allowed_registry> <1> <2>
+ - Insecure Registries: <insecure_registry> <3>
+ - Allowed Registries for Import: <4>
+    - Domain Name: <domain_name> <5>
+    - Insecure: true <6>
+ - Additional Trusted CA: <7>
+    - <ca_name> : REDACTED
```
Comment (chenz4027) [inline modules/images-editing-image-registry-settings-hcp.adoc:113]: should be <registry_name>

File: modules/images-configuration-image-registry-settings-hcp.adoc [hunk 3]
```diff
@@ -96,17 +96,21 @@ Audit Log Forwarding:       Disabled
...
-<3> `Allowed Registries for Import`: Limits the container image registries from which normal users can import images. The format should be a comma-separated list of `domainName:insecure`.
-<4> `domainName`: Specifies a domain name for the registry.
-<5> `insecure`: Indicates whether the registry is secure or insecure.
+<2> `Blocked Registries`: A comma-separated list of registries for which image pull and push actions are blocked. Parameters `allowedRegistries`, `blockedRegistries` are mutually exclusive.
+<3> `Insecure Registries`: A comma-separated list of registries which do not have a valid TLS certificate or only support HTTP connections.
+<4> `Allowed Registries for Import`: Limits the container image registries from which normal users can import images. The format should be a comma-separated list of `domainName:insecure`.
+<5> `domainName`: Specifies a domain name for the registry.
+<6> `insecure`: Indicates whether the registry is secure or insecure.
+<7> `Additional Trusted CA`: A JSON file containing the registry hostname as the key, and the PEM-encoded certificate as the value, for each additional registry CA to trust.
```
Comment (yingzhanredhat) [inline modules/images-configuration-image-registry-settings-hcp.adoc:113]: It should be same with cluster edition.

File: modules/images-editing-image-registry-settings-hcp.adoc [hunk 4]
```diff
@@ -29,7 +29,8 @@ You can change any registry-related parameter, which will trigger a rollout acro
 $ rosa edit cluster --registry-config-insecure-registries <insecure_registries> \
    --registry-config-allowed-registries <allowed_registries> \
    --registry-config-allowed-registries-for-import <registry_name:insecure> \
-   --registry-config-additional-trusted-ca <additional_trusted_ca_file>
+   --registry-config-additional-trusted-ca <additional_trusted_ca_file> \
+   --registry-config-platform-allowlist <platform_allowlist_id>
 ----
```
[the remark this answers is not in the document]
Comment (fmcdonal) [reply modules/images-editing-image-registry-settings-hcp.adoc:34]: removed

File: modules/images-configuration-image-registry-settings-hcp.adoc (hunk 3 above)
Comment (fmcdonal) [reply modules/images-configuration-image-registry-settings-hcp.adoc]: applied

File: modules/images-configuration-image-registry-settings-hcp.adoc [hunk 5]
```diff
@@ -96,17 +96,25 @@ Audit Log Forwarding:       Disabled
...
-<4> `domainName`: Specifies a domain name for the registry.
-<5> `insecure`: Indicates whether the registry is secure or insecure.
+<2> `Blocked Registries`: A comma-separated list of registries for which image pull and push actions are blocked. Parameters `allowedRegistries`, `blockedRegistries` are mutually exclusive.
+<3> `Insecure Registries`: A comma-separated list of registries which do not have a valid TLS certificate or only support HTTP connections.
+<4> `Allowed Registries for Import`: Limits the container image registries from which normal users can import images. The format should be a comma-separated list of `domainName:insecure`.
+<5> `domainName`: Specifies a domain name for the registry.
+<6> `insecure`: Indicates whether the registry is secure or insecure.
+<7> `Platform Allowlist`: A reference to the id of the list of registries that needs to be whitelisted for the platform to work. It can be omitted at creation and updating and its lifecycle can be managed separately if needed.
```
Comment (chenz4027) [inline modules/images-configuration-image-registry-settings-hcp.adoc:115]: We should not show platform allowlist for create, it's never supported.

Comment (fmcdonal) [reply modules/images-configuration-image-registry-settings-hcp.adoc:115]: removed

File: modules/images-configuration-image-registry-settings-hcp.adoc (hunk 3 above)
Comment (yingzhanredhat) [reply modules/images-configuration-image-registry-settings-hcp.adoc:113]: The Platform Allowlist item is not shown in cluster description  after cluster creation

File: modules/images-editing-image-registry-settings-hcp.adoc [hunk 6]
```diff
@@ -104,14 +104,22 @@ Audit Log Forwarding:       Disabled
...
-<4> `domainName`: Specifies a domain name for the registry.
-<5> `insecure`: Indicates whether the registry is secure or insecure.
\ No newline at end of file
+<2> `Blocked Registries`: A comma-separated list of registries for which image pull and push actions are blocked. Parameters `allowedRegistries`, `blockedRegistries` are mutually exclusive.
+<3> `Insecure Registries`: A comma-separated list of registries which do not have a valid TLS certificate or only support HTTP connections.
+<4> `Allowed Registries for Import`: Limits the container image registries from which normal users can import images. The format should be a comma-separated list of `domainName:insecure`.
+<5> `domainName`: Specifies a domain name for the registry.
+<6> `insecure`: Indicates whether the registry is secure or insecure.
+<7> `Platform Allowlist`: A reference to the id of the list of registries that needs to be whitelisted for the platform to work. It can be omitted at creation and updating and its lifecycle can be managed separately if needed.
```
Comment (andreadecorte) [inline modules/images-editing-image-registry-settings-hcp.adoc:123]:
```suggestion
<7> `Platform Allowlist`: A reference to the id of the list of registries that needs to be whitelisted for the platform to work.
```

**Review by EricPonvelle — COMMENTED** [review]
/lgtm

**Pull request merged**